src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as path from "path";
+import vscode, { Uri } from "vscode";
+import {
+  LanguageClient,
+  TextDocumentIdentifier,
+} from "vscode-languageclient/node";
+
+import {
+  AzExtTreeDataProvider,
+  IActionContext,
+  IAzureQuickPickItem,
+  parseError,
+  UserCancelledError,
+} from "@microsoft/vscode-azext-utils";
+
+import {
+  BicepDeployParams,
+  bicepDeployRequestType,
+  deploymentScopeRequestType,
+} from "../language";
+import { AzLoginTreeItem } from "../tree/AzLoginTreeItem";
+import { AzResourceGroupTreeItem } from "../tree/AzResourceGroupTreeItem";
+import { LocationTreeItem } from "../tree/LocationTreeItem";
+import { TreeManager } from "../tree/TreeManager";
+import { OutputChannelManager } from "../utils/OutputChannelManager";
+import { Command } from "./types";
+
+export class DeployCommand implements Command {
+  public readonly id = "bicep.deploy";
+
+  public constructor(
+    private readonly client: LanguageClient,
+    private readonly outputChannelManager: OutputChannelManager,
+    private readonly treeManager: TreeManager
+  ) {}
+
+  public async execute(
+    _context: IActionContext,
+    documentUri?: vscode.Uri | undefined
+  ): Promise<void> {
+    documentUri ??= vscode.window.activeTextEditor?.document.uri;
+
+    if (!documentUri) {
+      return;
+    }
+
+    if (documentUri.scheme === "output") {
+      // The output panel in VS Code was implemented as a text editor by accident. Due to breaking change concerns,
+      // it won't be fixed in VS Code, so we need to handle it on our side.
+      // See https://github.com/microsoft/vscode/issues/58869#issuecomment-422322972 for details.
+      vscode.window.showInformationMessage(
+        "Unable to locate an active Bicep file, as the output panel is focused. Please focus a text editor first before running the command."
+      );
+
+      return;
+    }
+
+    const documentPath = documentUri.fsPath;
+    const textDocument = TextDocumentIdentifier.create(documentUri.fsPath);
+    const fileName = path.basename(documentPath);
+    this.outputChannelManager.appendToOutputChannel(
+      `Started deployment of ${fileName}`
```

Comment:
Hmmm if the deployment fails because of errors that were caught by our validation engine, then we show:
![image](https://user-images.githubusercontent.com/30270536/158886426-0ed981be-83cf-4fde-af4c-3262cb05f205.png)

For scenario where build at our side passes fine, I am not sure we need to log additional messages. Would it be of use for  end user to know processing details? I don't feel strongly about it. If @ucheNkadiCode agrees I am happy to change.

Reply:
k